### PR TITLE
Remove TestRunner::installStatisticsDidScanDataRecordsCallback

### PR DESCRIPTION
**Pull request by achristensen07**
#### 5bb70059d37c2f0ef7b26cf2e26be06054eb5572
<pre>
Remove TestRunner::installStatisticsDidScanDataRecordsCallback
<a href="https://bugs.webkit.org/show_bug.cgi?id=278508">https://bugs.webkit.org/show_bug.cgi?id=278508</a>
<a href="https://rdar.apple.com/134462116">rdar://134462116</a>

Reviewed by Pascoe.

It stores state in the injected bundle and doesn&apos;t work with site isolation.
With some reordering callbacks after awaits, we can get the functionality to be tested
at equivalent times.  When it&apos;s gone, we can remove a lot of piping needed to support it.

* LayoutTests/http/tests/resourceLoadStatistics/cookie-deletion-expected.txt:
* LayoutTests/http/tests/resourceLoadStatistics/cookie-deletion.html:
* LayoutTests/http/tests/resourceLoadStatistics/prune-statistics.html:
* LayoutTests/http/tests/resourceLoadStatistics/user-interaction-in-cross-origin-sub-frame.html:
* LayoutTests/http/tests/webAPIStatistics/canvas-read-and-write-data-collection.html:
* LayoutTests/http/tests/webAPIStatistics/font-load-data-collection.html:
* LayoutTests/http/tests/webAPIStatistics/navigator-functions-accessed-data-collection.html:
* LayoutTests/http/tests/webAPIStatistics/screen-functions-accessed-data-collection.html:
* LayoutTests/http/tests/websocket/connection-refusal-in-frame-resource-load-statistics.html:
* LayoutTests/http/tests/websocket/resources/localhost-websocket-connect.html:
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp:
(WebKit::ResourceLoadStatisticsStore::processStatisticsAndDataRecords):
(WebKit::ResourceLoadStatisticsStore::grandfatherExistingWebsiteData):
(WebKit::ResourceLoadStatisticsStore::setNotifyPagesWhenDataRecordsWereScanned): Deleted.
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.h:
* Source/WebKit/NetworkProcess/Classifier/WebResourceLoadStatisticsStore.cpp:
(WebKit::WebResourceLoadStatisticsStore::resourceLoadStatisticsUpdated):
(WebKit::WebResourceLoadStatisticsStore::registrableDomainsWithWebsiteData):
(WebKit::WebResourceLoadStatisticsStore::setNotifyPagesWhenDataRecordsWereScanned): Deleted.
(WebKit::WebResourceLoadStatisticsStore::notifyResourceLoadStatisticsProcessed): Deleted.
* Source/WebKit/NetworkProcess/Classifier/WebResourceLoadStatisticsStore.h:
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::registrableDomainsWithWebsiteData):
(WebKit::NetworkProcess::setNotifyPagesWhenDataRecordsWereScanned): Deleted.
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/NetworkProcess.messages.in:
* Source/WebKit/NetworkProcess/NetworkSession.cpp:
(WebKit::NetworkSession::registrableDomainsWithWebsiteData):
(WebKit::NetworkSession::notifyResourceLoadStatisticsProcessed): Deleted.
* Source/WebKit/NetworkProcess/NetworkSession.h:
* Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreRef.cpp:
(WKWebsiteDataStoreSetStatisticsNotifyPagesWhenDataRecordsWereScanned): Deleted.
* Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreRef.h:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::setNotifyPagesWhenDataRecordsWereScanned): Deleted.
(WebKit::NetworkProcessProxy::notifyResourceLoadStatisticsProcessed): Deleted.
(WebKit::NetworkProcessProxy::notifyWebsiteDataScanForRegistrableDomainsFinished): Deleted.
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.messages.in:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::notifyPageStatisticsAndDataRecordsProcessed): Deleted.
(WebKit::WebProcessProxy::notifyWebsiteDataScanForRegistrableDomainsFinished): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::setNotifyPagesWhenDataRecordsWereScanned): Deleted.
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebResourceLoadObserver.cpp:
(WebKit::WebResourceLoadObserver::scheduleNotificationIfNeeded):
(WebKit::WebResourceLoadObserver::updateCentralStatisticsStore):
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::didReceiveMessageToPage):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::installStatisticsDidScanDataRecordsCallback): Deleted.
(WTR::TestRunner::statisticsDidScanDataRecordsCallback): Deleted.
(WTR::TestRunner::setStatisticsNotifyPagesWhenDataRecordsWereScanned): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::setStatisticsNotifyPagesWhenDataRecordsWereScanned): Deleted.
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveSynchronousMessageFromInjectedBundle):

Canonical link: <a href="https://commits.webkit.org/282626@main">https://commits.webkit.org/282626@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/07a21878f4d5841d52d3d9a1fc277285c75b6d0e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63656 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/43012 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16253 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67677 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14264 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65776 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/50809 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14544 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51289 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9861 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66725 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39898 "Unexpected infrastructure issue, retrying build") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55106 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31919 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36580 "Unexpected infrastructure issue, retrying build") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12485 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13137 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/58328 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12810 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69373 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7603 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12377 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58564 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7636 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55199 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58783 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6325 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9635 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38833 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39912 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/41024 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39655 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/WebProcess/WebCoreSupport/WebResourceLoadObserver.cpp [hunk 1]
```diff
@@ -104,13 +104,16 @@ void WebResourceLoadObserver::scheduleNotificationIfNeeded()
         return;
     }
 
-    if (!m_notificationTimer.isActive())
+    if (!m_notificationTimer.isActive()) {
+        WTFReportBacktrace();
```
Comment (rreno) [inline Source/WebKit/WebProcess/WebCoreSupport/WebResourceLoadObserver.cpp]: Debug logging?

File: Source/WebKit/WebProcess/WebCoreSupport/WebResourceLoadObserver.cpp [hunk 2]
```diff
@@ -104,13 +104,16 @@ void WebResourceLoadObserver::scheduleNotificationIfNeeded()
...
         m_notificationTimer.startOneShot(minimumNotificationInterval);
+    }
 }
 
 void WebResourceLoadObserver::updateCentralStatisticsStore(CompletionHandler<void()>&& completionHandler)
 {
     m_notificationTimer.stop();
+    WTFLogAlways("WebResourceLoadObserver::updateCentralStatisticsStore called");
```
Comment (rreno) [inline Source/WebKit/WebProcess/WebCoreSupport/WebResourceLoadObserver.cpp]: Debug logging?

File: Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp [hunk 3]
```diff
@@ -457,6 +451,7 @@ void ResourceLoadStatisticsStore::removeDataRecords(CompletionHandler<void()>&&
 
 void ResourceLoadStatisticsStore::processStatisticsAndDataRecords()
 {
+    WTFReportBacktrace();
```
Comment (rreno) [inline Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsStore.cpp]: Debug logging?

File: Source/WebKit/NetworkProcess/Classifier/WebResourceLoadStatisticsStore.cpp [hunk 4]
```diff
@@ -315,6 +285,7 @@ void WebResourceLoadStatisticsStore::statisticsDatabaseHasAllTables(CompletionHa
 
 void WebResourceLoadStatisticsStore::resourceLoadStatisticsUpdated(Vector<ResourceLoadStatistics>&& statistics, CompletionHandler<void()>&& completionHandler)
 {
+    WTFReportBacktrace();
```
Comment (rreno) [inline Source/WebKit/NetworkProcess/Classifier/WebResourceLoadStatisticsStore.cpp]: Debug logging?

**Review by rreno — COMMENTED** [review]
I think you left some debug logging in the commit.

File: LayoutTests/http/tests/websocket/resources/localhost-websocket-connect.html [hunk 5]
```diff
@@ -1,4 +1,7 @@
 <!DOCTYPE html>
 <script>
-    new WebSocket('ws://localhost/');
+    w = new WebSocket('ws://localhost/');
+    w.onerror = ()=>{
```
Comment (pascoej) [inline LayoutTests/http/tests/websocket/resources/localhost-websocket-connect.html:4]: minuscule nit: spacing

**Review by pascoej — APPROVED** [review]

File: LayoutTests/http/tests/websocket/resources/localhost-websocket-connect.html (hunk 5 above)
Comment (achristensen07) [reply LayoutTests/http/tests/websocket/resources/localhost-websocket-connect.html:4]: I like this spacing!

**Review by charliewolfe — APPROVED** [review]

**Pull request merged**